### ISSUE TITLE
Bundle FSharpx.Collections' LazyList

### ISSUE THREAD

**Issue by moodmosaic** [jystic/dotnet-jack#46]
into `dotnet-jack` codebase using Paket's neat support for [*GitHub* dependencies](https://fsprojects.github.io/Paket/github-dependencies.html).

This way we can bundle `LazyList.fs` and `LazyList.fsi` into `dotnet-jack` during `./build.sh` and so #45 can be a whole lot easier. 

Now that https://github.com/fsprojects/FSharpx.Collections/pull/67 is merged, I have already something locally that seems to work, and looks like this:

```diff
diff --git a/Jack.Tests/Jack.Tests.fsproj b/Jack.Tests/Jack.Tests.fsproj
index d407817..753a507 100644
--- a/Jack.Tests/Jack.Tests.fsproj
+++ b/Jack.Tests/Jack.Tests.fsproj
@@ -97,17 +97,6 @@
   </Target>
   -->
   <Choose>
-    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And ($(TargetFrameworkVersion) == 'v4.0' Or $(TargetFrameworkVersion) == 'v4.5' Or $(TargetFrameworkVersion) == 'v4.5.1' Or $(TargetFrameworkVersion) == 'v4.5.2' Or $(TargetFrameworkVersion) == 'v4.5.3' Or $(TargetFrameworkVersion) == 'v4.6' Or $(TargetFrameworkVersion) == 'v4.6.1' Or $(TargetFrameworkVersion) == 'v4.6.2' Or $(TargetFrameworkVersion) == 'v4.6.3')">
-      <ItemGroup>
-        <Reference Include="FSharpx.Collections">
-          <HintPath>..\packages\FSharpx.Collections\lib\net40\FSharpx.Collections.dll</HintPath>
-          <Private>True</Private>
-          <Paket>True</Paket>
-        </Reference>
-      </ItemGroup>
-    </When>
-  </Choose>
-  <Choose>
     <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.0'">
       <ItemGroup>
         <Reference Include="Unquote">

diff --git a/Jack.Tests/paket.references b/Jack.Tests/paket.references
index 09cabf4..46dd121 100644
--- a/Jack.Tests/paket.references
+++ b/Jack.Tests/paket.references
@@ -2,4 +2,3 @@ xunit.runner.console
 xunit.core
 Unquote
 
-FSharpx.Collections

diff --git a/Jack/Jack.fsproj b/Jack/Jack.fsproj
index ac7e5ac..39efba8 100644
--- a/Jack/Jack.fsproj
+++ b/Jack/Jack.fsproj
@@ -45,6 +45,14 @@
     <Reference Include="System.Numerics" />
   </ItemGroup>
   <ItemGroup>
+    <Compile Include="..\paket-files\fsprojects\FSharpx.Collections\src\FSharpx.Collections\LazyList.fsi">
+      <Paket>True</Paket>
+      <Link>paket-files/LazyList.fsi</Link>
+    </Compile>
+    <Compile Include="..\paket-files\fsprojects\FSharpx.Collections\src\FSharpx.Collections\LazyList.fs">
+      <Paket>True</Paket>
+      <Link>paket-files/LazyList.fs</Link>
+    </Compile>
     <Compile Include="Numeric.fs" />
     <Compile Include="Seed.fs" />
     <Compile Include="Tree.fs" />
@@ -58,15 +66,4 @@
     <Compile Include="Script.fsx" />
   </ItemGroup>
   <Import Project="$(FSharpTargetsPath)" />
-  <Choose>
-    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And ($(TargetFrameworkVersion) == 'v4.0' Or $(TargetFrameworkVersion) == 'v4.5' Or $(TargetFrameworkVersion) == 'v4.5.1' Or $(TargetFrameworkVersion) == 'v4.5.2' Or $(TargetFrameworkVersion) == 'v4.5.3' Or $(TargetFrameworkVersion) == 'v4.6' Or $(TargetFrameworkVersion) == 'v4.6.1' Or $(TargetFrameworkVersion) == 'v4.6.2' Or $(TargetFrameworkVersion) == 'v4.6.3')">
-      <ItemGroup>
-        <Reference Include="FSharpx.Collections">
-          <HintPath>..\packages\FSharpx.Collections\lib\net40\FSharpx.Collections.dll</HintPath>
-          <Private>True</Private>
-          <Paket>True</Paket>
-        </Reference>
-      </ItemGroup>
-    </When>
-  </Choose>
-</Project>
+</Project>
\ No newline at end of file

diff --git a/Jack/Shrink.fs b/Jack/Shrink.fs
index fc096d0..24fda13 100644
--- a/Jack/Shrink.fs
+++ b/Jack/Shrink.fs
@@ -20,6 +20,9 @@ module LazyList =
             else
                 LazyList.cons x <| LazyList.cons y ys
 
+module private List =
+    let inline cons hd tl = hd::tl
+
 module Shrink =
     /// Permutes a list by removing 'k' consecutive elements from it:
     ///

diff --git a/Jack/paket.references b/Jack/paket.references
index 89cf281..a24c8a2 100644
--- a/Jack/paket.references
+++ b/Jack/paket.references
@@ -1 +1,2 @@
-FSharpx.Collections
+File: LazyList.fs
+File: LazyList.fsi
\ No newline at end of file

diff --git a/paket.dependencies b/paket.dependencies
index f716e48..559aa22 100644
--- a/paket.dependencies
+++ b/paket.dependencies
@@ -4,7 +4,8 @@ source https://www.nuget.org/api/v2/
 nuget FAKE 4.41.1
 nuget xunit.runner.console 2.1.0
 
-nuget FSharpx.Collections 1.15.2
+github fsprojects/FSharpx.Collections:944b8c8e464cdd10070889f53c78e9fbeb478698 src/FSharpx.Collections/LazyList.fs
+github fsprojects/FSharpx.Collections:944b8c8e464cdd10070889f53c78e9fbeb478698 src/FSharpx.Collections/LazyList.fsi
 
 nuget xunit.core 2.1.0
 nuget Unquote 3.1.2

diff --git a/paket.lock b/paket.lock
index 10eca13..e9e32d7 100644
--- a/paket.lock
+++ b/paket.lock
@@ -2,9 +2,6 @@ REDIRECTS: ON
 NUGET
   remote: https://www.nuget.org/api/v2
     FAKE (4.41.1)
-    FSharp.Core (4.0.0.1)
-    FSharpx.Collections (1.15.2)
-      FSharp.Core
     Microsoft.NETCore.Platforms (1.0.1) - framework: dnxcore50
     Microsoft.NETCore.Targets (1.0.1) - framework: dnxcore50
     System.Collections (4.0.11) - framework: dnxcore50
@@ -133,3 +130,7 @@ NUGET
       xunit.abstractions (>= 2.0) - framework: dnxcore50
       xunit.extensibility.core (2.1) - framework: >= net45, dnx451, dnxcore50, monoandroid, monotouch, xamarinios, winv4.5, wpv8.0, wpav8.1
     xunit.runner.console (2.1)
+GITHUB
+  remote: fsprojects/FSharpx.Collections
+    src/FSharpx.Collections/LazyList.fs (944b8c8e464cdd10070889f53c78e9fbeb478698)
+    src/FSharpx.Collections/LazyList.fsi (944b8c8e464cdd10070889f53c78e9fbeb478698)
\ No newline at end of file
```

**Comment by jystic** [comment]
So does this mean that if we distribute on NuGet we'll be bundling LazyList inside our assembly and we won't need the dependency?

That sounds pretty good 👍

**Comment by moodmosaic** [comment]
Exactly 😀

**Comment by moodmosaic** [comment]
Resolved by #48.
